From ca537c255a096a1ebd385dabaf27d54f6369ce99 Mon Sep 17 00:00:00 2001
From: Tristan Cacqueray <tdecacqu@redhat.com>
Date: Tue, 11 Jul 2017 03:04:23 +0000
Subject: [PATCH] Better handle pika connection errors

It seems like pika may raise AttributeError when the connection is closed,
as describe here: https://github.com/pika/pika/issues/638, resulting
in this error displayed on the webclient:
500: POST /storyboard_api/tasks: 'NoneType' object has no attribute 'send'

This change adds AttributeError to the ConnectionClosed except.

Change-Id: Iffa1ef1998379097b32d699a360b5de9277c01ed
---

diff --git a/storyboard/notifications/publisher.py b/storyboard/notifications/publisher.py
index 1b5c8b0..3425497 100644
--- a/storyboard/notifications/publisher.py
+++ b/storyboard/notifications/publisher.py
@@ -103,7 +103,7 @@
             if payload in self._pending:
                 self._pending.remove(payload)
             return True
-        except ConnectionClosed as cc:
+        except (ConnectionClosed, AttributeError) as cc:
             LOG.warning(_LW("Attempted to send message on closed connection."))
             LOG.debug(cc)
             self._open = False
